### DISCUSSION TITLE
conditionally disable inlayHints

### DISCUSSION THREAD

**Discussion by Harry-Lees** [microsoft/pylance-release#3180] [Enhancement]
The inlayHints feature is really helpful, and I've been using both variable, and function hints since they came out. However, there are some cases where the inlayHints add visual clutter rather than useful information. The biggest example of this can be seen in Enums and constants:

![image](https://user-images.githubusercontent.com/52263746/183885145-f344e85b-6daa-4852-b545-7af77dac3cfc.png)

![image](https://user-images.githubusercontent.com/52263746/183885281-a3c48457-4186-4e28-b75c-5646c49e4420.png)

It would be nice if there were a way to disable `Literal[]` hints since it seems to be a corner case, Or if there were some way of disabling inlayHints on a specific block of code so large enums, or a method with a large number of declarations e.g., a large `__init__()` method could be disabled.

**Comment by heejaechang** [comment]
I am +1 on not showing inlay hint when it is very obvious. but there are some cases Literal is useful such as when there is constant folding 

ex) a = 1 + 1
inlay hint will show Literal[2]

or literal with different format 
ex) a = 0xA103

it would be nice if inlay hint adds a bit more logics around this and only hide very obvious one. I think TS has some of this logics.

**Reply by patrys** [reply]
That's a good point, but if I had to choose between the two, I'd easily pick having no literal annotations over having annotations on all literals as we have now.

**Reply by erictraut** [reply]
We documented some guidelines for library authors that indicate [which symbols do not need annotations](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#type-completeness). These symbols are guaranteed to be inferred consistently by all Python type checkers. I recommend using this list of guidelines to determine whether it's appropriate to show inlay hints. The list includes enum members, constants (i.e. a symbol that is unconditionally assigned a literal value in only one place), and type aliases.

**Comment by MrYutz** [comment]
The Enum inlay hints are very distracting.  So much so that I ended up here.....

Is there a flag to ignore just those lines?  Similar to # type: ignore

So someone else can find this discussion with less searching, here is the VS Code settings flag for inlay hints:
`python.analysis.inlayHints.variableTypes`

**Comment by heejaechang** [comment]
tagging @bschnurr
